pulser/register.py

```diff
@@ -282,6 +282,29 @@ def _draw_checks(
                     "Needs more than one atom to draw " "the blockade radius."
                 )
 
+    @abstractmethod
+    def _to_dict(self) -> dict[str, Any]:
+        qs = dict(zip(self._ids, map(np.ndarray.tolist, self._coords)))
+        return obj_to_dict(self, qs)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, BaseRegister):
+            return False
+
+        return (
+            type(self) is type(other)
+            and set(self._ids) == set(other._ids)
+            and all(
+                (
+                    np.array_equal(
+                        self._coords[i],
+                        other._coords[other._ids.index(self._ids[i])],
+                    )
+                    for i in range(len(self._ids))
+                )
```

Comment:
I would loop over the `_ids` instead:
```suggestion
                (
                    np.array_equal(
                        self._coords[i],
                        other._coords[other._ids.index(id)],
                    )
                    for i, id in enumerate(self._ids)
                )
```